src/main/java/uk/gov/hmcts/ccd/domain/service/common/AccessControlService.java

```diff
@@ -329,8 +321,76 @@ private boolean hasCreateAccess(String newFieldName, final List<CaseField> caseF
+            return false;
+        }
+        return itemUpdatedAndHasUpdateAccess(existingData, newData, newFieldName, caseFieldDefinitions, userRoles);
+    }
+
+    private boolean itemDeletedAndHasDeleteAccess(JsonNode existingData, JsonNode newData, String newFieldName,
+                                                  List<CaseField> caseFieldDefinitions, Set<String> userRoles) {
+        boolean containsDeletedItem = StreamSupport
+            .stream(spliteratorUnknownSize(existingData.get(newFieldName).elements(), Spliterator.ORDERED), false)
+            .anyMatch(oldItem -> itemMissing(oldItem, newData.get(newFieldName)));
+
+        return !containsDeletedItem || hasCaseFieldAccess(caseFieldDefinitions, userRoles, CAN_DELETE, newFieldName);
+    }
+
+    private boolean itemMissing(JsonNode oldItem, JsonNode newValue) {
+        boolean itemExists = StreamSupport.stream(spliteratorUnknownSize(newValue.elements(), Spliterator.ORDERED), false)
+            .anyMatch(newItem -> !isNullId(newItem) && newItem.get("id").equals(oldItem.get("id")));
+        return !itemExists;
+    }
+    
+    private boolean itemUpdatedAndHasUpdateAccess(JsonNode existingData, JsonNode newData, String newFieldName,
+                                                  List<CaseField> caseFieldDefinitions, Set<String> userRoles) {
+        boolean containsUpdatedItem = StreamSupport
+            .stream(spliteratorUnknownSize(existingData.get(newFieldName).elements(), Spliterator.ORDERED), false)
+            .anyMatch(oldItem -> itemUpdated(oldItem, newData.get(newFieldName)));
+
+        return !containsUpdatedItem || hasCaseFieldAccess(caseFieldDefinitions, userRoles, CAN_UPDATE, newFieldName);
+    }
+
+    private boolean itemUpdated(JsonNode oldItem, JsonNode newValue) {
+        return StreamSupport.stream(spliteratorUnknownSize(newValue.elements(), Spliterator.ORDERED), false)
+            .anyMatch(newItem -> {
+                boolean itemExists = !isNullId(newItem) && newItem.get("id").equals(oldItem.get("id"));
+                if (itemExists) {
+                    return !newItem.equals(oldItem);
+                }
+                return false;
+            });
+    }
+
+    private boolean itemAddedAndHasCreateAccess(JsonNode newData, String newFieldName, List<CaseField> caseFieldDefinitions, Set<String> userRoles) {
+        JsonNode newValue = newData.get(newFieldName);
+        boolean containsNewItem = StreamSupport.stream(spliteratorUnknownSize(newValue.elements(), Spliterator.ORDERED), false)
+            .anyMatch(newItem -> isNullId(newItem));
+        return !containsNewItem || hasCreateAccess(newFieldName, caseFieldDefinitions, userRoles);
+    }
+
+    private boolean isNullId(JsonNode newItem) {
+        return newItem.get("id") == null
+            || newItem.get("id").equals(NullNode.getInstance())
+            || newItem.get("id").asText().equalsIgnoreCase("null");
     }
 
```

Comment:
If it were protected, it would be a lot simpler to unit test, wouldn't it

Reply:
Actually added 4 more test cases which are testing this method indirectly... Not sure is it's a good idea to change scope just for the sake of testing

Reply:
ok